### DISCUSSION TITLE
Translations (i18n)

### DISCUSSION THREAD

**Discussion by varna** [ream/ream#198] [Ideas]
Hey,

I wanted to discuss about few things concerning languages.

### How do we localize?

> **Locale** is a code of ISO 639-1  or combination of ISO 639-1 and  ISO 3166-1 alpha-2, with a hyphen (i.e. 'en-US').

Currently, most common and recommended approach is to add **locale** as first param: `/[locale]/[...otherParams]`.

Nuxt provides us with: https://i18n.nuxtjs.org/

### How do we deal with locale list?

My locale list in most cases is a dynamic response from API (Translation platform). And that's why I need to write custom functions for:
- meta/head translations and canonical links
- sitemap.xml
- generate/export static pages

### How do we import/fetch translations?

There are three types of translations that I'm interested in:
- Page - translations needed for current page
- Component - translations needed for current component, i.e. list of items for `select` element.
- Global - globally imported translations available everywhere, also known as "Translation Project" for single App. Could also be seen as App Shell form Google or Template in Nuxt.

### How can we translate links?

It would be nice to be able to translate URLs like this:
```
/[locale]/[category]/[product].vue
/en/fridges/snowflake-3000
/lt/šaldytuvai/snaigė-3000
```

**Comment by ghost** [comment]
Interested to know too. How i18n story goes with ream?
